src/main/java/org/testng/internal/BaseTestMethod.java

```diff
@@ -624,11 +629,27 @@ public IRetryAnalyzer getRetryAnalyzer() {
     return m_retryAnalyzer;
   }
 
+  @Override
+  public IRetryAnalyzer getRetryAnalyzer(ITestResult result) {
+    return getRetryAnalyzerConsideringMethodParameteters(result);
+  }
+
   @Override
   public void setRetryAnalyzer(IRetryAnalyzer retryAnalyzer) {

```

Comment:
you should throw an exception if the method is called because it is not supposed to be the case

Reply:
@juherr - That basically is as good as us removing the methods without deprecation. I believe the whole notion of deprecation is to just warn users that an API is going to be removed but still work. Throwing exceptions will cause the functionality to cease working and need code change by downstream users. If that's what we need then why use a deprecation strategy ? I am not sure I get this.